clang/lib/Headers/hlsl/hlsl_intrinsic_helpers.h

```diff
@@ -0,0 +1,71 @@
+//===----- hlsl_intrinsic_helpers.h - HLSL helpers intrinsics -------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _HLSL_HLSL_INTRINSIC_HELPERS_H_
```

Comment:
How does the `enable_if_t` template in  `hlsl_detail.h`get exposed here?

Reply:
hlsl_detail.h is included first in hlsl.h, which makes it included before the other headers. These headers are all implementation details, and aren't expected to be exposed to user code since hlsl.h is implicitly included in all HLSL source files.